File: ws/tipantiza/u2/ws13ArrayJson/Store/main_store/model/Product.js
class Product {

    constructor(id, name, category, price) {

        this.id = id;
        this.name = name;
        this.category = category;
        this.price = price;
    }

    calculateSubtotal(quantity) {
        return this.price * quantity;
    }

    calculateTax(subtotal) {
        return subtotal * 0.12;
    }

    calculateTotal(subtotal, tax) {
        return subtotal + tax;
    }
}

module.exports = Product;

File: ws/tipantiza/u2/ws13ArrayJson/Store_Python/model/Product.py
class Product:
    def __init__(self, product_id: int, name: str, category: str, price: float):
        self.id = product_id
        self.name = name
        self.category = category
        self.price = price

    def calculate_subtotal(self, quantity: int) -> float:
        return self.price * quantity

    def calculate_tax(self, subtotal: float) -> float:
        return subtotal * 0.12

    def calculate_total(self, subtotal: float, tax: float) -> float:
        return subtotal + tax